Compile errors with latest ADAM 0.23.0-SNAPSHOT version

```
...
[ERROR] cannoli/src/main/scala/org/bdgenomics/cannoli/Bwa.scala:199: error: value copy
is not a member of org.bdgenomics.adam.rdd.read.AlignmentRecordRDD
[ERROR] possible cause: maybe a semicolon is missing before `value copy'?
[ERROR]      .copy(recordGroups = RecordGroupDictionary(Seq(RecordGroup(sample, sample))))
[ERROR]       ^
[ERROR] cannoli/src/main/scala/org/bdgenomics/cannoli/Bwa.scala:203: error: value copy
is not a member of org.bdgenomics.adam.rdd.read.AlignmentRecordRDD
[ERROR]      output.copy(sequences = sequences)
[ERROR]             ^
[ERROR] cannoli/src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala:32:
error: class FragmentRDD is abstract; cannot be instantiated
[ERROR]    new FragmentRDD(apply(rdd.rdd),
[ERROR]    ^
[ERROR] three errors found
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

--- a/pom.xml
+++ b/pom.xml
@@ -361,6 +361,11 @@
         </exclusions>
       </dependency>
       <dependency>
+        <groupId>org.apache.spark</groupId>
+        <artifactId>spark-sql_${scala.version.prefix}</artifactId>
+        <version>${spark.version}</version>
+      </dependency>
+      <dependency>
         <groupId>org.scalatest</groupId>
         <artifactId>scalatest_${scala.version.prefix}</artifactId>
         <version>2.2.6</version>
@@ -371,6 +376,11 @@
     <dependency>
       <groupId>org.apache.spark</groupId>
       <artifactId>spark-core_${scala.version.prefix}</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-sql_${scala.version.prefix}</artifactId>
       <scope>provided</scope>
     </dependency>
     <dependency>

--- a/src/main/scala/org/bdgenomics/cannoli/Bwa.scala
+++ b/src/main/scala/org/bdgenomics/cannoli/Bwa.scala
@@ -196,11 +196,11 @@ class Bwa(protected val args: BwaArgs) extends BDGSparkCommand[BwaArgs] with Log
     }
 
     val output: AlignmentRecordRDD = input.pipe[AlignmentRecord, AlignmentRecordRDD, InterleavedFASTQInFormatter](bwaCommand)
-      .copy(recordGroups = RecordGroupDictionary(Seq(RecordGroup(sample, sample))))
+      .replaceRecordGroups(RecordGroupDictionary(Seq(RecordGroup(sample, sample))))
 
     val outputMaybeWithSequences = Option(args.sequenceDictionary).fold(output)(sdPath => {
       val sequences = SequenceDictionaryReader(sdPath, sc)
-      output.copy(sequences = sequences)
+      output.replaceSequences(sequences)
     })
 
     if (!args.asFragments) {

--- a/src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala
+++ b/src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala
@@ -29,9 +29,8 @@ private[cannoli] object QuerynameGrouper extends Serializable {
   }
 
   def apply(rdd: AlignmentRecordRDD): FragmentRDD = {
-    new FragmentRDD(apply(rdd.rdd),
+    FragmentRDD(apply(rdd.rdd),
       rdd.sequences,
-      rdd.recordGroups,
-      optPartitionMap = None)
+      rdd.recordGroups)
   }
 }

